app/server_inactivity.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/mattermost/mattermost-server/v6/store"
+)
+
+const serverInactivityHours = 100
+
+func (s *Server) doInactivityCheck() {
+	if !s.Config().FeatureFlags.EnableInactivityCheckJob {
+		mlog.Info("No activity check because EnableInactivityCheckJob feature flag is disabled")
+		return
+	}
+
+	inactivityDurationHourseEnv := os.Getenv("MM_INACTIVITY_DURATION")
+	inactivityDurationHours, parseError := strconv.ParseFloat(inactivityDurationHourseEnv, 64)
+	if parseError != nil {
+		// default to 100 hours
+		inactivityDurationHours = serverInactivityHours
+	}
+
+	systemValue, sysValErr := s.Store.System().GetByName("INACTIVITY")
+	if sysValErr != nil {
+		if _, ok := sysValErr.(*store.ErrNotFound); !ok {
+			mlog.Warn("An error occurred while getting INACTIVITY from system store", mlog.Err(sysValErr))
+		}
+	}
+
+	// If we have a system value, it means this job already ran atleast once.
+	// we then check the last time the job ran plus the last time a post was made to determine if we
+	// can remind the user to use workspace again. If no post was made, we check the last time they logged in (session)
+	// and determine whether to send them a reminder.
+	if systemValue != nil {
+		sysT, _ := strconv.ParseInt(systemValue.Value, 10, 64)
+		tt := time.Unix(sysT/1000, 0)
+		timeLastSentInativityEmail := time.Since(tt).Hours()
+
+		lastPostAt, _ := s.Store.Post().GetLastPostRowCreateAt()
+		if lastPostAt != 0 {
+			posT := time.Unix(lastPostAt/1000, 0)
+			timeForLastPost := time.Since(posT).Hours()
+
+			if timeLastSentInativityEmail > inactivityDurationHours && timeForLastPost > inactivityDurationHours {
+				s.takeInactivityAction()
+			}
+			return
+		}
+
+		lastSessionAt, _ := s.Store.Session().GetLastSessionRowCreateAt()
+		if lastSessionAt != 0 {
+			sesT := time.Unix(lastSessionAt/1000, 0)
+			timeForLastSession := time.Since(sesT).Hours()
+
+			if timeLastSentInativityEmail > inactivityDurationHours && timeForLastSession > inactivityDurationHours {
+				s.takeInactivityAction()
+			}
+			return
+		}
+	}
+
+	// The first time this job runs. We check if the user has not made any posts
+	// and remind them to use the workspace. If no posts have been made. We check the last time
+	// they logged in (session) and send a reminder.
+
+	lastPostAt, _ := s.Store.Post().GetLastPostRowCreateAt()
+	if lastPostAt != 0 {
+		posT := time.Unix(lastPostAt/1000, 0)
+		timeForLastPost := time.Since(posT).Hours()
+		if timeForLastPost > inactivityDurationHours {
+			s.takeInactivityAction()
+		}
+		return
+	}
+
+	lastSessionAt, _ := s.Store.Session().GetLastSessionRowCreateAt()
+	if lastSessionAt != 0 {
+		sesT := time.Unix(lastSessionAt/1000, 0)
+		timeForLastSession := time.Since(sesT).Hours()
+		if timeForLastSession > inactivityDurationHours {
+			s.takeInactivityAction()
+		}
+		return
+	}
+}
+
+func (s *Server) takeInactivityAction() {
+	siteURL := *s.Config().ServiceSettings.SiteURL
+	properties := map[string]interface{}{
+		"SiteURL": siteURL,
+	}
```

Comment:
Do we need to default this to something if no SiteURL is set?

Reply:
Since SiteURL is critical to the telemetry and sending of the email, we can consider this an error
```go
if siteURL == "" {
		mlog.Error("SiteURL needs to be set to run inactivity job")
		return
	}
```